tools/inference/torch_inference.py

```diff
@@ -85,7 +85,8 @@ def infer(args: Namespace) -> None:
 
         if args.output is None and args.show is False:
             warnings.warn(
```

Comment:
Can we replace all warnings.warn to `logger.warn`? This way we will be able to redirect them to a file if needed and use rich's console logging to print the warnings.